examples/tests/test_mnist.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+from examples.tests import runner
+
+
+EXAMPLES_ROOT = runner.EXAMPLES_ROOT
+
+
+def test_mnist():
+    template_dir = os.path.join(
+        EXAMPLES_ROOT, 'mnist/.testdata/templates')
+
+    output_evaluator = runner.TemplateOutputEvaluator(
+        b'''\
+Device: @numpy
+# unit: 10
+# Minibatch-size: 100
+# epoch: 1
+
+epoch       main/loss   validation/main/loss  main/accuracy  validation/main/accuracy  elapsed_time
+1           {a        } {b                  } {c           } {d                      } {e         }
+''',  # NOQA
+        a=(float, lambda x: 0.6 < x < 1.5),
+        b=(float, lambda x: 0.3 < x < 0.6),
+        c=(float, lambda x: 0.62 < x < 0.82),
+        d=(float, lambda x: 0.83 < x < 0.98),
+        e=(float, lambda x: 0. < x < 100.),
+    )
```

Comment:
I would like to find information for choosing this new format method.
1) Possible other solutions
2) Pros and cons of these solutions (and thus why the above code is good)

Reply:
Actually I didn't come up with any other reasonable solutions.
I'll try to list possibilities any way.

- Write in each test a logic to check the output line by line

  - Pros: What's being done in each step is clear from test implementation.
  - Cons: Prohibitively difficult to write/maintain test cases.

- Use a regular expression to capture the numbers, and use lambdas to check the numerical ranges.

  - Pros: Being a standard notation
  - Cons: Difficult to express the output with a regular expression. It's also not easy to interpret for human. Possibilities to not be able to express with the syntax.